src/backend/distributed/commands/function.c

```diff
@@ -187,22 +190,36 @@ create_distributed_function(PG_FUNCTION_ARGS)
 	EnsureFunctionOwner(funcOid);
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
-	ErrorIfFunctionDependsOnExtension(&functionAddress);
 
 	/*
-	 * when we allow propagation within a transaction block we should make sure to only
-	 * allow this in sequential mode
+	 * If the function is owned by an extension, only update the
+	 * pg_dist_object, and not propagate the CREATE FUNCTION. Function
+	 * will be created by the virtue of the extension creation.
 	 */
-	EnsureSequentialModeForFunctionDDL();
+	if (IsObjectAddressOwnedByExtension(&functionAddress, &extensionAddress))
```

Comment:
Do we also need to cover the citus_add_node scenario? (adding a node when an extension-owned function is distributed)

Reply:
You meant unit test the scenario? if So, we are doing that in src/test/regress/sql/propagate_extension_commands.sql at line:280

![image](https://user-images.githubusercontent.com/44680808/152696919-f789d07f-5193-4196-bf9e-70ff4be06240.png)

Reply:
If I do a citus_add_node after using create_distributed_function on an extension-owned function, the citus_add_node seems to propagate CREATE OR REPLACE FUNCTION command, while we explicitly skip that at create_distributed_function time. We should probably address that, since recreating the function might not be allowed.

Reply:
Changed the code in ReplicateAllObjectsToNodeCommandList(). Verified manually, adding DEBUG messages is the best but it's showing too many unrelated messages